test/CoreTest.cpp

```diff
@@ -102,8 +102,11 @@ TEST_CASE( "output_constructor_test", "[core]" )
 
     o.setMeshesPath("customMeshesPath").setParticlesPath("customParticlesPath");
 
-    o.iterations[1].meshes["foo"];
-    o.iterations[1].particles["bar"];
+    o.iterations[1].meshes["foo"]["baz"].resetDataset(Dataset(Datatype::DOUBLE, {1}));
```

Comment:
can you add a (global) extend-zero write to the tests and it's defined behaviour? Basically `empty_record_test` but where the user sets the extend explicitly to `{0}`?

Do we already have a test that does a parallel write with extend > 0 (1D and ND) in which some of the MPI ranks contribute *locally* zero extends? This is quite often the case for e.g. particle data.

Reply:
Ok, great. Just one thing:

> If the extent passed to resetDataset() is zero in any dimension (and is later flushed to disk), undefined behaviour in certain backends ensures.

Instead of undef behaviour, depending on backend, can we just implement defined similar behaviour for all backends? E.g., if global zero extent is given we throw (directly or in flush) or if zero is given we skip the write (and document that), etc.?

Reply:
> if global zero extent is given we throw (directly or in flush)

That's the behaviour I currently implement in #309. Skipping the undefined write (however verbose) does not appeal to me, as the dataset itself does not really make sense.